### ISSUE TITLE
Combine two kinds of loss function

### ISSUE THREAD

**Issue by wang5566** [Cysu/open-reid#45]
I want to train my model using soft max
Loss and triplet loss together.  How can I combine two loss in a net together? Any reply will be appreciated.

**Comment by Rizhiy** [comment]
You need to write a new loss, similar to the one `reid/loss/triplet.py` and use that instead. Usually, you can just add losses together, just make sure to scale them since softmax can be much larger than triplet.

You would probably need to write your own data loader since softmax and triplet use different sampling techniques.

**Comment by Cysu** [comment]
@Rizhiy Thank you very much for the answer!

**Comment by wang5566** [comment]
@Rizhiy Thank you for your reply. But how to scale this two loss? In my network, I got softmax loss 1.5 and triplet loss 1.27, seeming unusual.

**Comment by Rizhiy** [comment]
To be honest I have never tried to train a network with multi-loss. Perhaps train softmax and triplet loss separately first, then calculate the scale such that they become the same value.

**Comment by zydou** [comment]
You can add a parameter to control the balance between them.
For example:
```
loss = loss1 + lambda * loss2 
```

**Comment by Cysu** [comment]
@Rizhiy @zydou Thanks a lot for your answers!
